lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+
+if (!inspector || !inspector.asyncTaskScheduled) {
+  exports.setup = function() {};
+  return;
+}
+
+const hook = createHook({
+  init(asyncId, type, triggerId, resource) {
```

Comment:
Just to confirm that it's alright with you, while we make sure calls to `init()` in core actually mean the event loop has progressed in some way since the call to `init()`, the user API allows users to call `before()` synchronously. There isn't much that can be done about that, and it's on the user to implement it properly.

Basically, there's nothing actionable. Just want to bring this point up.